### DISCUSSION TITLE
web browser examples

### DISCUSSION THREAD

**Discussion by unicomp21** [dotnet/Silk.NET#1300] [Q&A] [answered]
Is there a good source of examples for running in the browser on webgpu?

**Comment by Beyley** [comment, marked as answer]
we do not support WebGPU in the browser

**Reply by thygrrr** [reply]
What about in 2025?

**Comment by twodawg** [comment]
Can we get a WebGPU example? It doesn't have to be in the browser. Thank you!

**Comment by clibequilibrium** [comment]
> Can we get a WebGPU example? It doesn't have to be in the browser. Thank you!

You can check Lab folder you have 3 native examples there https://github.com/dotnet/Silk.NET/tree/main/src/Lab/Experiments/WebGPUTriangle

Hope it helps !

**Reply by Perksey** [reply]
Note that these are not examples, anything in src/Lab often demonstrates terrible coding practice and is likely extremely bug prone, so it’s probably better if you just pretend those don’t exist.

**Reply by clibequilibrium** [reply]
I don't think it matters in this context. a 300 lines webgpu sample that can be refactored by OP is better that not knowing where to start at all. I am in fact thankful for Lab folder because it has a lot of good use cases for Silk.NET , otherwise Silk is unusable for beginners if your logic is applied.

**Reply by clibequilibrium** [reply]
Also it is partially true. for example. in sample of ImGui and Vulkan, the code can be refactored but it is perfectly functional and runs on Mac/Windows/Linux and helped me bootstrap a good gpu driven rendering with C# and Silk.NET

**Reply by Beyley** [reply]
> I don't think it matters in this context. a 300 lines webgpu sample that can be refactored by OP is better that not knowing where to start at all. I am in fact thankful for Lab folder because it has a lot of good use cases for Silk.NET , otherwise Silk is unusable for beginners if your logic is applied.

the WebGPU code in the lab is very bad, i can attest as i am the one who wrote it, it forgets to delete a bunch of stuff and shows sub-optimal use of the API, id reccomend looking at the `wgpu-native` samples to see how to use the API, its not in C#, but our bindings are 1:1, so it should be about the same
https://github.com/gfx-rs/wgpu-native/tree/trunk/examples

the webgpu lab code is basically just a "smoke test" for the bindings
